### PR TITLE
Feature/initial flow

### PR DESCRIPTION
**Pull request by dvaleda**
Implementation of choosing the best time for lunch and sending that data to the agent CultureMaster.

### CODE REVIEW

File: services/rule_engine/src/main/java/signals/DayChoiceSignal.java [hunk 1]
```diff
@@ -1,13 +1,17 @@
 package signals;
 
-import java.util.ArrayList;
-
 import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import java.util.ArrayList;
```
Comment (assert-0) [inline services/rule_engine/src/main/java/signals/DayChoiceSignal.java]: Fix import order.

File: services/rule_engine/src/main/java/agents/Ava.java [hunk 2]
```diff
@@ -20,8 +22,50 @@ public Ava(String connectionName, String connectionId) {
...
+        String description = Mitems.getText("weekly-core.description-asking-for-available-days.text");
+        com.mindsmiths.mitems.Option[] days = Mitems.getOptions("weekly-core.days.each-day");
+        
+        List<CloudSelectComponent.Option> options = List.of(
+            new CloudSelectComponent.Option(days[0].getText(), "0", true),
+            new CloudSelectComponent.Option(days[1].getText(), "1", true),
+            new CloudSelectComponent.Option(days[2].getText(), "2", true),
+            new CloudSelectComponent.Option(days[3].getText(), "3", true),
+            new CloudSelectComponent.Option(days[4].getText(), "4", true)
+        );
```
Comment (mjuraj) [inline services/rule_engine/src/main/java/agents/Ava.java]: Use for loop or something

File: services/rule_engine/src/main/java/agents/Ava.java [hunk 3]
```diff
@@ -20,8 +22,50 @@ public Ava(String connectionName, String connectionId) {
...
+            .addComponent("title", new TitleComponent(title))
+            .addComponent("text", new DescriptionComponent(description))
+            .addComponent("cloudSelect", new CloudSelectComponent("availableDays", options))
+            .addComponent("submit", new PrimarySubmitButtonComponent("submit", "confirmDays"));
+        showScreen(daysScreen);
+    }
+
+    public void confirmingDaysScreen() {
+        String title1 = Mitems.getText("weekly-core.confirmation-of-choosen-available-days.title");
+        String title2 = Mitems.getText("weekly-core.stay-tuned-second-confirmation-of-available-days.title");
```
Comment (mjuraj) [inline services/rule_engine/src/main/java/agents/Ava.java]: Directly use Mitems in addComponent

File: services/rule_engine/src/main/java/signals/DayChoiceSignal.java [hunk 4]
```diff
@@ -2,12 +2,16 @@
...
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
 
 @AllArgsConstructor
+@NoArgsConstructor
+@Data
 public class DayChoiceSignal extends Signal {
-    ArrayList<Integer> freeDays;    
+    ArrayList<String> notFreeDays;
```
Comment (mjuraj) [inline services/rule_engine/src/main/java/signals/DayChoiceSignal.java]: List

File: services/rule_engine/src/main/resources/rules/Ava/weeklyCore.drl [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+import java.util.*;
+
+import com.mindsmiths.armory.events.UserConnectedEvent
+import com.mindsmiths.armory.events.SubmitEvent
+
+import agents.Ava;
+import agents.CultureMaster;
+import signals.DayChoiceSignal;
+
+rule "Best Time"
```
Comment (mjuraj) [inline services/rule_engine/src/main/resources/rules/Ava/weeklyCore.drl]: "Ask for available days"

File: services/rule_engine/src/main/resources/rules/Ava/weeklyCore.drl [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    when
+        signal: UserConnectedEvent() from entry-point "signals"
+        agent: Ava()
+    then 
+        agent.chooseAvailableDaysScreen();
+        delete(signal);
+end
+
+rule "Confirm days"
```
Comment (mjuraj) [inline services/rule_engine/src/main/resources/rules/Ava/weeklyCore.drl]: What happens when 0 days are selected

**Pull request closed without merging**